### ISSUE TITLE
Departmental permissions—requirements

### ISSUE THREAD

**Issue by desigonz** [cityofaustin/techstack#3378]
### As a Joplin user, I want permissions to be tied to my department, so I have access to content that is relevant to me.

Once #3227 is implemented, let's turn @toribr's requirements below into MVP acceptance criteria (maybe a table similar to the #3227 criteria) and then review with content and dev.

We'll need to think about this in terms of the editor/moderator/administrator flow. (Admin probably can see all, no departmental restrictions; editors can only see content for their department; but what about moderators?)
- **Editor** is the most basic role; can create and save drafts, but can’t publish
- **Moderators** can publish
- **Admins** can do everything

Things to consider:
- What is the experience of an admin assigning a Joplin user to a department? Will need mocks/acceptance criteria.
- We might need to make decisions around topics/topic collections. Do editors have access to topics/topic collections?
- Snippets, images, and documents aren't assigned by department. Can a user see all of them?

### As an editor within a particular department, I want to only see content that is relevant to me, so that I can more easily access my content.

- [ ] As a city author, I would only be able to view/edit pages and content (documents/images) that belong to my department.
- [ ] In the case of guides, I need to be able to access pages from other departments via the page picker 

Assumptions:
 - [ ] Each user will have a profile, which includes the department they work in, their name,
and their email
- [ ] When a user logs in, they will see only content on "pages" that has been created by themselves or those within their department. Similarly they will only see documents and images uploaded by themselves or someone from their department.
- Adding pages to a guide from another department is an exception: when users go to add pages by default they will see only pages created by themselves or others in their department. If the user selects the checkbox, however, the page should refresh to show all eligible pages across Alpha.
![#2912](https://user-images.githubusercontent.com/3790067/66419412-b7a12380-e9c9-11e9-81ad-7efc4d1317ee.png)

### As an editor within a particular department, I don't want to fill out fields with information that the system should already know about me.

- [ ] When creating a new page, "offered by" would automatically be assigned to pages, so no department assignment would need to be made manually. 
- [ ] The department selection on page templates remains as is for now. For future versions, we should change the presentation of this selection.

## Parking for now
Things to consider:
- Future state, we’re thinking only moderators and admins will be assigning pages to the IA. 

### As an author, I only want to see topic and topic collections within my department so I don't have to view topics that aren't relevant to my content.
- [ ] When authors go to select a topic for new pages they should only see topics from their department.

Notes (not synthesized) https://docs.google.com/document/d/154RQ3aIliIjcyk_Wd2MD74wo4OAHq1Xt9GjL8ik4o2Y/edit#
- Re: Topic control from Mass.gov https://massgovdigital.gitbook.io/knowledge-base/content-types-1/topic-pages/about-topic-pages
 
Currently topics aren't associated with departments. However, in order to prevent the average user from viewing topics that aren't relevant, we should constrain the view of topics to those that are currently associated with pages created by that department. 

- [ ] Similar to #3085, this checkbox allows viewing of all topics across Alpha.

![#3124](https://user-images.githubusercontent.com/3790067/66425803-1d93a800-e9d6-11e9-8d09-8e8791641255.png)

**Comment by desigonz** [comment]
## MVP requirements: early thoughts
To review at content pin-up and dev-design sync on 12/10
- [ ] Editors and moderators must belong to at least one department
  - [ ] **Design team** will need to create designs for how an administrator assigns a department to a department when [adding a new user](https://joplin.herokuapp.com/admin/users/)—maybe on the roles page?
  - [ ] Question for **dev team**: should department pages = permissions for departments, or should we separate those in the data model?
- [ ] On the main content search page, editors and moderators only see pages that belong to their assigned department 
- [ ] When using a page picker, editors and moderators have the ability to search and select all pages, regardless of assigned department

## Probably will be post-MVP
- [ ] When using a page picker, editors and moderators can filter out pages that belong/don't belong to their assigned department
- [ ] When an editor or a moderator creates a new page, the department(s) they are assigned to are automatically selected on the content page's department picker
  - Question for **design and content**: I wonder if we should wait on this until we do more research about the editorial flow (and maybe understand how Drupal currently has departments implemented). Will there be instances in which moderators might have access to multiple departments' content? How would a department auto-appear on a content page in that instance?

**Comment by desigonz** [comment]
@annieszot I'd like to review this at the next content pin-up—maybe 5-10 minutes on the agenda?

**Comment by desigonz** [comment]
- [ ] @desigonz to make dev requirements issue and maybe work with dev to split it up

**Comment by desigonz** [comment]
Logged these all as new issues: #3780, #3781, #3782, #3784.
